### PR TITLE
Bump Sphinx version to 1.5.5

### PR DESCRIPTION
**Pull request by pv**
Update scipy-sphinx-theme to fix some appearance issues with new Sphinx versions.
Bump CI to use Sphinx 1.5.5.

### CODE REVIEW

**Comment by pv** [comment]
circleci: https://2691-1460385-gh.circle-artifacts.com/0/home/ubuntu/scipy/doc/build/html-scipyorg/index.html

File: .travis.yml [hunk 1]
```diff
@@ -112,7 +112,7 @@ before_install:
   - if [ "${USE_WHEEL}" == "1" ]; then pip install wheel; fi
   - |
     if [ "${REFGUIDE_CHECK}" == "1" ]; then
-        travis_retry pip install matplotlib Sphinx==1.2.3
+        travis_retry pip install matplotlib Sphinx==1.5.5
```
Comment (Eric89GXL) [inline .travis.yml]: Why do we force a specific version? Not doing so would ensure future version compatibility.

Comment (pv) [inline .travis.yml]: I think to avoid random breakage of the builds due to sphinx bugs.

**Comment by ev-br** [comment]
Thanks Pauli

**Comment by pv** [comment]
This apparently broke the refguide check. Need to fix that...

**Pull request merged**